addons/calendar/calendar.py

```diff
@@ -1440,7 +1441,13 @@ def get_interval(self, cr, uid, ids, date, interval, tz=None, context=None):
         elif interval == 'time':
             dummy, format_time = self.get_date_formats(cr, uid, context=context)
             res = date.strftime(format_time + " %Z")
-        return res
+        return res.decode(getpreferredencoding(do_setlocale=False))
```

Comment:
In terms of bytestring decoding, the proper and simpler way to convert an arbitrary bytestring to unicode is to use `tools.ustr()`, in our Python2 landscape.

However it seems to me that having the result in the machine locale is simply wrong, there is no guarantee it will match the user's locale. In general we should not depend on `datetime` for locale-specific formatting, and use `babel.dates.format_date[time]` instead, so we can pass the user's locale as appropriate (as done in [read_group](https://github.com/odoo/odoo/blob/8.0/openerp/models.py#L2016) for instance, after format conversion).
Babel's formatting methods seem to return safe unicode strings, so there should be less problems for this too.